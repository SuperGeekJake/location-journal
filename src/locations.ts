import { sub, set } from "date-fns";

import { MurmurHash3, Xoshiro128 } from "./random";

export interface IVisit {
  id: number;
  locationId: number;
  date: number;
}

export interface ILocation {
  id: number;
  name: string;
  tags: string[];
}

// Use seeded random to create predictable locations
const gs = MurmurHash3("Olivia loves BTS!");
const random = Xoshiro128(gs(), gs(), gs(), gs());

const randomInt = (max: number, min: number = 1) =>
  Math.floor(random() * max) + min;

const randomBoolean = (chance: number) => random() < chance;

const names = [
  "Aroma Mocha",
  "Grinders Cafe",
  "Steamy Beans Coffee",
  "Baristas",
  "Ground Up Cafe",
  "Tatiana's Cafe",
  "Beans 'n Cream Cafe",
  "HuggaMug Cafe",
  "The Busy Bean",
  "Beany Business",
  "Impresso Espresso",
  "The Coffee Club",
  "Boston Barista",
  "Jacked Up Coffee",
  "The Family Bean",
  "Club Coffee",
  "Jumpin' Beans Cafe",
  "The Friendly Bean",
  "Coffee Express",
  "Jumpstart Coffee Shop",
  "The Grind",
  "Coffee House",
  "Lava Java",
  "The Hideout",
  "Coffee Time",
  "Manhattan Mocha",
  "The Roasted Bean",
  "Cup o' Joe",
  "Mugs Coffee",
  "The Split Bean",
  "Dream Bean Coffee Shop",
  "No Doze Cafe",
  "The Steam Room",
  "Espresso Cafe",
  "Screamin' Beans",
  "Topped Off",
  "Espresso Express",
  "Spiced Chai Cafe",
  "Wake Up Cafe",
  "Expresso",
  "Steam Beans Cafe",
  "Wide Awake Cafe",
  "Grind House",
  "Steamin' Mugs",
  "Yo Jo Coffee Shop",
];

export const locations = names.map<ILocation>((name, index) => ({
  id: index + 1,
  name,
  tags: randomBoolean(0.3) ? ["featured"] : [],
}));

export const locationMap = locations.reduce<Record<number, ILocation>>(
  (result, location) => {
    result[location.id] = location;
    return result;
  },
  {}
);

const dates = Array.from({ length: 100 }, (_, index) =>
  sub(
    set(new Date(), {
      hours: randomInt(23, 0),
      minutes: randomInt(59, 0),
      seconds: randomInt(59, 0),
      milliseconds: randomInt(999, 0),
    }),
    { days: randomInt(index + 5, index) }
  ).getTime()
).sort((a, b) => a - b);

export const visits = dates
  .map<IVisit>((date, index) => ({
    id: index + 1,
    locationId: randomInt(locations.length),
    date,
  }))
  .reverse();
