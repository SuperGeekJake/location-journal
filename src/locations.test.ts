import { describe, it, expect } from "vitest";
import * as data from "./locations";

describe("Fabricated data", () => {
  it("locations should match snapshot", () => {
    expect(data.locations).toMatchSnapshot();
  });

  it("visits should match snapshot", () => {
    expect(data.visits).toMatchSnapshot();
  });
});
