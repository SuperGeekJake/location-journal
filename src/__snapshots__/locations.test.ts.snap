// Vitest Snapshot v1

exports[`Fabricated data > locations should match snapshot 1`] = `
[
  {
    "id": 1,
    "name": "Aroma Mocha",
    "tags": [
      "featured",
    ],
  },
  {
    "id": 2,
    "name": "Grinders Cafe",
    "tags": [
      "featured",
    ],
  },
  {
    "id": 3,
    "name": "Steamy Beans Coffee",
    "tags": [],
  },
  {
    "id": 4,
    "name": "Baristas",
    "tags": [],
  },
  {
    "id": 5,
    "name": "Ground Up Cafe",
    "tags": [],
  },
  {
    "id": 6,
    "name": "Tatiana's Cafe",
    "tags": [],
  },
  {
    "id": 7,
    "name": "Beans 'n Cream Cafe",
    "tags": [
      "featured",
    ],
  },
  {
    "id": 8,
    "name": "HuggaMug Cafe",
    "tags": [],
  },
  {
    "id": 9,
    "name": "The Busy Bean",
    "tags": [],
  },
  {
    "id": 10,
    "name": "Beany Business",
    "tags": [
      "featured",
    ],
  },
  {
    "id": 11,
    "name": "Impresso Espresso",
    "tags": [],
  },
  {
    "id": 12,
    "name": "The Coffee Club",
    "tags": [],
  },
  {
    "id": 13,
    "name": "Boston Barista",
    "tags": [],
  },
  {
    "id": 14,
    "name": "Jacked Up Coffee",
    "tags": [],
  },
  {
    "id": 15,
    "name": "The Family Bean",
    "tags": [
      "featured",
    ],
  },
  {
    "id": 16,
    "name": "Club Coffee",
    "tags": [],
  },
  {
    "id": 17,
    "name": "Jumpin' Beans Cafe",
    "tags": [],
  },
  {
    "id": 18,
    "name": "The Friendly Bean",
    "tags": [
      "featured",
    ],
  },
  {
    "id": 19,
    "name": "Coffee Express",
    "tags": [],
  },
  {
    "id": 20,
    "name": "Jumpstart Coffee Shop",
    "tags": [],
  },
  {
    "id": 21,
    "name": "The Grind",
    "tags": [],
  },
  {
    "id": 22,
    "name": "Coffee House",
    "tags": [],
  },
  {
    "id": 23,
    "name": "Lava Java",
    "tags": [],
  },
  {
    "id": 24,
    "name": "The Hideout",
    "tags": [],
  },
  {
    "id": 25,
    "name": "Coffee Time",
    "tags": [],
  },
  {
    "id": 26,
    "name": "Manhattan Mocha",
    "tags": [],
  },
  {
    "id": 27,
    "name": "The Roasted Bean",
    "tags": [],
  },
  {
    "id": 28,
    "name": "Cup o' Joe",
    "tags": [
      "featured",
    ],
  },
  {
    "id": 29,
    "name": "Mugs Coffee",
    "tags": [],
  },
  {
    "id": 30,
    "name": "The Split Bean",
    "tags": [],
  },
  {
    "id": 31,
    "name": "Dream Bean Coffee Shop",
    "tags": [],
  },
  {
    "id": 32,
    "name": "No Doze Cafe",
    "tags": [],
  },
  {
    "id": 33,
    "name": "The Steam Room",
    "tags": [],
  },
  {
    "id": 34,
    "name": "Espresso Cafe",
    "tags": [
      "featured",
    ],
  },
  {
    "id": 35,
    "name": "Screamin' Beans",
    "tags": [
      "featured",
    ],
  },
  {
    "id": 36,
    "name": "Topped Off",
    "tags": [],
  },
  {
    "id": 37,
    "name": "Espresso Express",
    "tags": [],
  },
  {
    "id": 38,
    "name": "Spiced Chai Cafe",
    "tags": [],
  },
  {
    "id": 39,
    "name": "Wake Up Cafe",
    "tags": [],
  },
  {
    "id": 40,
    "name": "Expresso",
    "tags": [
      "featured",
    ],
  },
  {
    "id": 41,
    "name": "Steam Beans Cafe",
    "tags": [
      "featured",
    ],
  },
  {
    "id": 42,
    "name": "Wide Awake Cafe",
    "tags": [],
  },
  {
    "id": 43,
    "name": "Grind House",
    "tags": [],
  },
  {
    "id": 44,
    "name": "Steamin' Mugs",
    "tags": [
      "featured",
    ],
  },
  {
    "id": 45,
    "name": "Yo Jo Coffee Shop",
    "tags": [],
  },
]
`;

exports[`Fabricated data > visits should match snapshot 1`] = `
[
  {
    "date": 1666656330591,
    "id": 100,
    "locationId": 7,
  },
  {
    "date": 1666254225272,
    "id": 99,
    "locationId": 10,
  },
  {
    "date": 1666116372975,
    "id": 98,
    "locationId": 6,
  },
  {
    "date": 1666065724907,
    "id": 97,
    "locationId": 34,
  },
  {
    "date": 1665949428119,
    "id": 96,
    "locationId": 12,
  },
  {
    "date": 1665922680088,
    "id": 95,
    "locationId": 33,
  },
  {
    "date": 1665666174137,
    "id": 94,
    "locationId": 15,
  },
  {
    "date": 1665573640619,
    "id": 93,
    "locationId": 5,
  },
  {
    "date": 1665496367261,
    "id": 92,
    "locationId": 45,
  },
  {
    "date": 1665486269358,
    "id": 91,
    "locationId": 16,
  },
  {
    "date": 1665416076183,
    "id": 90,
    "locationId": 21,
  },
  {
    "date": 1665410887712,
    "id": 89,
    "locationId": 24,
  },
  {
    "date": 1665137044352,
    "id": 88,
    "locationId": 6,
  },
  {
    "date": 1665097366338,
    "id": 87,
    "locationId": 25,
  },
  {
    "date": 1665076241995,
    "id": 86,
    "locationId": 3,
  },
  {
    "date": 1664928189800,
    "id": 85,
    "locationId": 17,
  },
  {
    "date": 1664815465530,
    "id": 84,
    "locationId": 36,
  },
  {
    "date": 1664644417502,
    "id": 83,
    "locationId": 41,
  },
  {
    "date": 1664573660320,
    "id": 82,
    "locationId": 12,
  },
  {
    "date": 1664462381572,
    "id": 81,
    "locationId": 16,
  },
  {
    "date": 1664177385733,
    "id": 80,
    "locationId": 21,
  },
  {
    "date": 1664165894061,
    "id": 79,
    "locationId": 3,
  },
  {
    "date": 1664044264864,
    "id": 78,
    "locationId": 37,
  },
  {
    "date": 1664036448768,
    "id": 77,
    "locationId": 10,
  },
  {
    "date": 1663338117920,
    "id": 76,
    "locationId": 14,
  },
  {
    "date": 1663004267350,
    "id": 75,
    "locationId": 38,
  },
  {
    "date": 1662984270503,
    "id": 74,
    "locationId": 19,
  },
  {
    "date": 1662966892685,
    "id": 73,
    "locationId": 11,
  },
  {
    "date": 1662794616285,
    "id": 72,
    "locationId": 1,
  },
  {
    "date": 1662667047368,
    "id": 71,
    "locationId": 5,
  },
  {
    "date": 1662637150630,
    "id": 70,
    "locationId": 27,
  },
  {
    "date": 1662538803025,
    "id": 69,
    "locationId": 21,
  },
  {
    "date": 1662369902909,
    "id": 68,
    "locationId": 21,
  },
  {
    "date": 1662219256969,
    "id": 67,
    "locationId": 3,
  },
  {
    "date": 1662128607585,
    "id": 66,
    "locationId": 28,
  },
  {
    "date": 1662089115001,
    "id": 65,
    "locationId": 34,
  },
  {
    "date": 1662033537574,
    "id": 64,
    "locationId": 37,
  },
  {
    "date": 1661959000436,
    "id": 63,
    "locationId": 9,
  },
  {
    "date": 1661957113247,
    "id": 62,
    "locationId": 42,
  },
  {
    "date": 1661810072675,
    "id": 61,
    "locationId": 45,
  },
  {
    "date": 1661594983388,
    "id": 60,
    "locationId": 24,
  },
  {
    "date": 1661417421954,
    "id": 59,
    "locationId": 9,
  },
  {
    "date": 1661205264638,
    "id": 58,
    "locationId": 40,
  },
  {
    "date": 1661106367621,
    "id": 57,
    "locationId": 5,
  },
  {
    "date": 1660853067439,
    "id": 56,
    "locationId": 24,
  },
  {
    "date": 1660694719251,
    "id": 55,
    "locationId": 37,
  },
  {
    "date": 1660675030024,
    "id": 54,
    "locationId": 44,
  },
  {
    "date": 1660378687166,
    "id": 53,
    "locationId": 30,
  },
  {
    "date": 1660177707738,
    "id": 52,
    "locationId": 39,
  },
  {
    "date": 1660052212003,
    "id": 51,
    "locationId": 16,
  },
  {
    "date": 1660018024595,
    "id": 50,
    "locationId": 45,
  },
  {
    "date": 1660005126587,
    "id": 49,
    "locationId": 21,
  },
  {
    "date": 1659936004793,
    "id": 48,
    "locationId": 44,
  },
  {
    "date": 1659848521072,
    "id": 47,
    "locationId": 4,
  },
  {
    "date": 1659636346625,
    "id": 46,
    "locationId": 25,
  },
  {
    "date": 1659551767438,
    "id": 45,
    "locationId": 8,
  },
  {
    "date": 1659456253842,
    "id": 44,
    "locationId": 4,
  },
  {
    "date": 1659431511623,
    "id": 43,
    "locationId": 5,
  },
  {
    "date": 1659403323857,
    "id": 42,
    "locationId": 27,
  },
  {
    "date": 1659394960884,
    "id": 41,
    "locationId": 21,
  },
  {
    "date": 1659236773692,
    "id": 40,
    "locationId": 4,
  },
  {
    "date": 1659222492531,
    "id": 39,
    "locationId": 7,
  },
  {
    "date": 1659159408265,
    "id": 38,
    "locationId": 44,
  },
  {
    "date": 1659084733052,
    "id": 37,
    "locationId": 29,
  },
  {
    "date": 1658635994253,
    "id": 36,
    "locationId": 29,
  },
  {
    "date": 1658600544409,
    "id": 35,
    "locationId": 38,
  },
  {
    "date": 1658256880364,
    "id": 34,
    "locationId": 21,
  },
  {
    "date": 1658186758285,
    "id": 33,
    "locationId": 35,
  },
  {
    "date": 1658104644317,
    "id": 32,
    "locationId": 31,
  },
  {
    "date": 1658092595950,
    "id": 31,
    "locationId": 19,
  },
  {
    "date": 1657650750966,
    "id": 30,
    "locationId": 2,
  },
  {
    "date": 1657246364897,
    "id": 29,
    "locationId": 7,
  },
  {
    "date": 1656844969988,
    "id": 28,
    "locationId": 13,
  },
  {
    "date": 1656806948540,
    "id": 27,
    "locationId": 27,
  },
  {
    "date": 1656729267142,
    "id": 26,
    "locationId": 2,
  },
  {
    "date": 1656350211325,
    "id": 25,
    "locationId": 20,
  },
  {
    "date": 1656268786621,
    "id": 24,
    "locationId": 39,
  },
  {
    "date": 1656047814049,
    "id": 23,
    "locationId": 21,
  },
  {
    "date": 1655970801891,
    "id": 22,
    "locationId": 25,
  },
  {
    "date": 1655912896416,
    "id": 21,
    "locationId": 5,
  },
  {
    "date": 1655877188777,
    "id": 20,
    "locationId": 6,
  },
  {
    "date": 1655459117295,
    "id": 19,
    "locationId": 38,
  },
  {
    "date": 1655087256836,
    "id": 18,
    "locationId": 3,
  },
  {
    "date": 1654629708369,
    "id": 17,
    "locationId": 20,
  },
  {
    "date": 1654560099837,
    "id": 16,
    "locationId": 10,
  },
  {
    "date": 1654445679816,
    "id": 15,
    "locationId": 13,
  },
  {
    "date": 1654406862204,
    "id": 14,
    "locationId": 4,
  },
  {
    "date": 1654329352716,
    "id": 13,
    "locationId": 24,
  },
  {
    "date": 1654288171257,
    "id": 12,
    "locationId": 40,
  },
  {
    "date": 1653712306151,
    "id": 11,
    "locationId": 21,
  },
  {
    "date": 1653484033947,
    "id": 10,
    "locationId": 34,
  },
  {
    "date": 1653018267261,
    "id": 9,
    "locationId": 15,
  },
  {
    "date": 1652721055908,
    "id": 8,
    "locationId": 16,
  },
  {
    "date": 1652672061299,
    "id": 7,
    "locationId": 45,
  },
  {
    "date": 1652181243307,
    "id": 6,
    "locationId": 19,
  },
  {
    "date": 1652099232718,
    "id": 5,
    "locationId": 42,
  },
  {
    "date": 1651622658128,
    "id": 4,
    "locationId": 38,
  },
  {
    "date": 1651233608171,
    "id": 3,
    "locationId": 45,
  },
  {
    "date": 1651051526051,
    "id": 2,
    "locationId": 16,
  },
  {
    "date": 1650856588533,
    "id": 1,
    "locationId": 29,
  },
]
`;
